lib/superstore/adapters/jsonb_adapter.rb

```diff
@@ -0,0 +1,210 @@
+gem 'pg'
+require 'pg'
+
+module Superstore
+  module Adapters
+    class JsonbAdapter < AbstractAdapter
+      class QueryBuilder
+        def initialize(adapter, scope)
+          @adapter  = adapter
+          @scope    = scope
+        end
+
+        def to_query
+          [
+            "SELECT #{select_string} FROM #{@scope.klass.table_name}",
+            where_string,
+            order_string,
+            limit_string
+          ].delete_if(&:blank?) * ' '
```

Comment:
overall, there is a lot of string building going on here that uses a combination of string interpolation, `Array * separator` and `Array.join(separator)`.  The most efficient method would to have all of the functions that are part of query building populate an array and then finally concatenate the array together in one step using `Array.join`